### ISSUE TITLE
Add ability to run tests using GDB instead of LLDB

### ISSUE THREAD

**Issue by enlight** [enlight/dbgmits#13]
Replace the current `run-tests` grunt task with `run-gdb-tests` and `run-lldb-tests` tasks.

**Comment by enlight** [comment]
This issue was partly resolved in 5a732869c77b05b3ed6c152a52d942558794c18b
More work is required to get most tests to actually run in GDB-MI, specifically for local debugging need to tell GDB to use a separate terminal for the inferior (which first requires creating a master/slave pseudo-terminal, probably using pty.js).

**Comment by enlight** [comment]
As of cec54f49f05c6dac3ae2a1fad2921e5b2132fb1b tests that should be skipped when using GDB or LLDB need to be tagged using @skipOnGDB and/or @skipOnLLDB, e.g.

```
it("does something @skipOnGDB", () => { expect(doesSomething()).to.be.true });
```

**Comment by enlight** [comment]
Resolved in cb4cf5c641fb8f75488d9e07a69dfb3756a248bc
